quinn/src/endpoint.rs

```diff
@@ -254,33 +252,48 @@ impl<S> EndpointInner<S>
 where
     S: proto::crypto::Session + 'static,
 {
-    fn drive_recv(&mut self, cx: &mut Context, now: Instant) -> Result<bool, io::Error> {
+    fn drive_recv<'a>(&'a mut self, cx: &mut Context, now: Instant) -> Result<bool, io::Error> {
         let mut recvd = 0;
+        let mut metas = [RecvMeta::default(); BATCH_SIZE];
+        let mut iovs = MaybeUninit::<[IoSliceMut<'a>; BATCH_SIZE]>::uninit();
+        self.recv_buf
+            .chunks_mut(self.recv_buf.len() / BATCH_SIZE)
+            .enumerate()
+            .for_each(|(i, buf)| unsafe {
+                iovs.as_mut_ptr()
+                    .cast::<IoSliceMut>()
+                    .add(i)
+                    .write(IoSliceMut::<'a>::new(buf));
+            });
+        let mut iovs = unsafe { iovs.assume_init() };
         loop {
-            match self.socket.poll_recv(cx, &mut self.recv_buf) {
-                Poll::Ready(Ok((n, addr, ecn))) => {
-                    match self
-                        .inner
-                        .handle(now, addr, ecn, (&self.recv_buf[0..n]).into())
-                    {
-                        Some((handle, DatagramEvent::NewConnection(conn))) => {
-                            let conn = self.create_connection(handle, conn);
-                            if self.incoming_live {
-                                self.incoming.push_back(conn);
-                                if let Some(task) = self.incoming_reader.take() {
-                                    task.wake();
+            match self.socket.poll_recv(cx, &mut iovs, &mut metas) {
+                Poll::Ready(Ok(msgs)) => {
```

Comment:
Later on there is a `recvd += 1;`, which counts the number of socket receive calls. Should this now be adjusted by `msgs` to account for the fact that it can now receive more messages at once?